src/kixi/lapita/schemas.clj

```diff
@@ -1,39 +1,49 @@
 (ns kixi.lapita.schemas
   (:require [schema.core :as s]
-            [schema.coerce :as coerce]))
+            [schema.coerce :as coerce]
+            [schema-contrib.core :as c]
+            [schema.utils :as su]))
 
 ;; Create and valid schemas using Plumatic schemas (https://github.com/plumatic/schema)
-(defn make-ordered-ds-schema [col-vec]
-  {:column-names (mapv #(s/one (s/eq (first %)) (str (first %))) col-vec)
-   :columns (mapv #(s/one [(second %)] (format "col %s" (name (first %)))) col-vec)
-   s/Keyword s/Any})
 
-(defn make-row-schema
-  [col-schema]
-  (mapv (fn [s] (let [datatype (-> s :schema first)
-                      fieldname (:name s)]
-                  (s/one datatype fieldname)))
-        (:columns col-schema)))
+;; recently updated to return data even if inconsistent, but w/ extra metadata info
+(defn coerce-data-from-schema
+  "Performs the data coercion using a schema.
+   If the data can be coerced, it's returned w/ :validated metadata associated.
+   If there's a coercion error the non coerced data is returned associated w/ :errors
+   metadata."
+  [data schema]
+  (let [data-coercer (coerce/coercer schema coerce/string-coercion-matcher)]
+    (pmap (fn [d] (let [coerced (data-coercer d)]
+                    (if (su/error? coerced)
+                      (with-meta d {:data-coercion :errors})
+                      (with-meta coerced {:data-coercion :validated}))))
```

Comment:
Could we avoid using metadata here and just return a map with `:results` and `:errors`?